llvm/lib/Target/RISCV/RISCVInstrInfo.td

```diff
@@ -150,12 +151,26 @@ def SPMem : MemOperand<SP>;
 
 def GPRCMem : MemOperand<GPRC>;
 
```

Comment:
I guess this makes a bunch of similar strings in the final binary that can no longer be shared?

Reply:
That's my expectation, yes. TableGen crimes have consequences, it turns out.

Reply:
I wonder how much work it be to introduce "DiagnosticCode" to allow us to programmatically generate the string like we do now

Reply:
I thought these (materialized) strings from TableGen code will eventually be put as normal C strings, so in the final compiler binaries similar string will still be deduplicated. Though these strings wouldn't be deduplicated in the *.inc files generated from TableGen I believe (though I think we can somehow solve this by changing how TableGen generates this part of the code)

Reply:
@mshockwave the range strings will all have a similar beginning and end but the lower and upper bounds are part of the string and will be different values. Without tablegen the prefix is and suffix are separate strings and the lower and upper bounds are converted to strings at runtime.

Reply:
> range strings will all have a similar beginning and end

oh right, I thought we're only talking about fixed strings.

Reply:
> I wonder how much work it be to introduce "DiagnosticCode" to allow us to programmatically generate the string like we do now

This feels like overenginnering. I'm like down three side-rabbit-holes now, and going to retreat a bit.

I will undo the tablegen crimes, the C++ is better for craig's reasons, but where we have fixed strings this should make things easier.